dodola/tests/test_services.py

```diff
@@ -29,7 +30,8 @@ def _datafactory(x, start_time="1950-01-01"):
     return out
 
 
-def test_bias_correct_basic_call():
+@pytest.mark.parametrize("method", ["QDM", "BCSD"])
+def test_bias_correct_basic_call(method):
```

Comment:
This feels scattered to me because we have "method"-handling code as part of the `parametrize` decorator _and_ in the test body (the if/elif/else clause picking head/tail vals).

 Could consolidate this by moving the correct expected values into the `@pytest.mark.parametrize`, so it passes the input method as well as its corresponding goal values. This is a fairly common pattern.

Quickly drawing this into a rough sketch to show what I'm talking about:
```python
@pytest.mark.parametrize(
    "method,expected_head,expected_tail",
    [
        pytest.param("BCSD", np.array([-0.08129293, -0.07613746, -0.0709855, -0.0658377, -0.0606947]), np.array([0.0520793, 0.06581804, 0.07096781, 0.07612168, 0.08127902]), id="BCSD head/tail"),
        pytest.param("QDM", np.array([4.0, 4.008609, 4.0172155, 4.0258169, 4.0344106]), np.array([3.9655894, 3.9741831, 3.9827845, 3.991391, 4.0]), id="QDM head/tail")
    ]
)
def test_bias_correct_basic_call(method, expected_head, expected_tail):
    """Simple integration test of bias_correct service

    We can't just test for removal of bias here since quantile mapping
    and adding in trend are both components of bias correction,
    so testing head and tail values instead
    """
    # Setup input data.
    n_years = 10
    n = n_years * 365  # need daily data...

    # Our "biased model".
    model_bias = 2
    ts = np.sin(np.linspace(-10 * np.pi, 10 * np.pi, n)) * 0.5
    x_train = _datafactory((ts + model_bias))
    # True "observations".
    y_train = _datafactory(ts)
    # Yes, we're testing and training on the same data...
    x_test = x_train.copy(deep=True)

    output_key = "test_output"
    training_model_key = "x_train"
    training_obs_key = "y_train"
    forecast_model_key = "x_test"

    # Load up a fake repo with our input data in the place of big data and cloud
    # storage.
    fakestorage = FakeRepository(
        {
            training_model_key: x_train,
            training_obs_key: y_train,
            forecast_model_key: x_test,
        }
    )

    bias_correct(
        forecast_model_key,
        x_train=training_model_key,
        train_variable="fakevariable",
        y_train=training_obs_key,
        out=output_key,
        out_variable="fakevariable",
        method=method,
        storage=fakestorage,
    )

    np.testing.assert_almost_equal(
        fakestorage.storage[output_key]["fakevariable"].squeeze(drop=True).values[:5],
        expected_head,
    )
    np.testing.assert_almost_equal(
        fakestorage.storage[output_key]["fakevariable"].squeeze(drop=True).values[-5:],
        expected_tail,
    )
```

Reply:
@brews I love this idea, much cleaner. Thanks! Updating as you suggested.